reqwest-tracing/src/reqwest_otel_span_macro.rs

```diff
@@ -98,26 +99,26 @@
 /// [`default_on_request_end`]: crate::reqwest_otel_span_builder::default_on_request_end
 macro_rules! reqwest_otel_span {
     // Vanilla root span at default INFO level, with no additional fields
-    ($request:ident) => {
-        reqwest_otel_span!($request,)
+    (name=$name:expr, $request:ident) => {
+        reqwest_otel_span!(name=$name, $request,)
     };
     // Vanilla root span, with no additional fields but custom level
-    (level=$level:expr, $request:ident) => {
-        reqwest_otel_span!(level=$level, $request,)
+    (level=$level:expr, name=$name:expr, $request:ident) => {
+        reqwest_otel_span!(level=$level, name=$name, $request,)
     };
     // Root span with additional fields, default INFO level
-    ($request:ident, $($field:tt)*) => {
-        reqwest_otel_span!(level=$crate::reqwest_otel_span_macro::private::Level::INFO, $request, $($field)*)
+    (name=$name:expr, $request:ident, $($field:tt)*) => {
+        reqwest_otel_span!(level=$crate::reqwest_otel_span_macro::private::Level::INFO, name=$name, $request, $($field)*)
     };
     // Root span with additional fields and custom level
-    (level=$level:expr, $request:ident, $($field:tt)*) => {
+    (level=$level:expr, name=$name:expr, $request:ident, $($field:tt)*) => {
```

Comment:
Since some users might be relying on the otel name currently being `METHOD PATH` in their spans, can we keep the same default behaviour (with no name provided).
If there is a name provided, including with the OtelName extension, then this can take priority.

I'm happy to consider a new major version soon though and I think it's reasonable to enforce this feature then

Reply:
I can do this, but I highly recommend not: otel systems don't deal particularly well with high cardinality for `name` - it messes with their UI's, search indices and so on. Given that reqwest-middleware is still a 0.x release, and this will:
- still have the URL on the span (and this full attribute searches will find it - and its where the tooling ecosystem expect it)
- not break compiles

I would recommend just changing it.

Reply:
I agree it should be the correct default, but I'd rather we allow for progressive upgrading rather than publish a breaking change

Reply:
Actually, it's fine. We're gonna make this a breaking change in the end